build.gradle

```diff
@@ -69,6 +70,15 @@ spotbugsTest {
 ext {
     opensearchVersion = '1.0.0'
     isSnapshot = "true" == System.getProperty("build.snapshot", "true")
+
+    // Name of the default git branch. This is used for checking out the RCA repository
+    gitDefaultBranch = 'main'
```

Comment:
Is there an explanation somewhere why this is needed? 

This should probably be parameterized, e.g. `-Dperformance-analyzer-rca.branch=main` and I would rename `cloneRcaProject` to `-Dperformance-analyzer-rca.build=false` or something like that to be consistent.

As a developer I probably also want to be able to `-Dperformance-analyzer-rca.path=...` instead.

Reply:
+1 
If RCA is always built as part of the assembled plugin zip for performance-analyzer, we will need the ability to define the branch that is needed for a particular release.  This works fine with main but will be needed to correctly reproduce builds from tags/branches from both repos.

Reply:
Alternative - It sounds like the RCA zip should be included as a maven publication that is fetched and used to build the final bundle in this repo rather than cloning & rebuilding or looking on a local path.

Reply:
I have modified the parameter name and added additional parameter ```performance-analyzer-rca.branch``` that helps to specify the RCA branch for checkout (defaults to main). 

Answering to why this is different from other types of dependencies. Currently, for creating the PA jar we need some files and folders from the RCA package, we use gradle for building and bring in the RCA bits. 
https://github.com/opendistro-for-elasticsearch/performance-analyzer/pull/131